cmd/teleporter-cli/transaction.go

```diff
@@ -30,46 +33,105 @@ var transactionCmd = &cobra.Command{
 	Short: "Parses relevant Teleporter logs from a transaction",
 	Long: `Given a transaction this command looks through the transaction's receipt
 for Teleporter and Warp log events. When corresponding log events are found,
-the command parses to log event fields to a more human readable format.`,
+the command parses to log event fields to a more human readable format. Optionally pass -d 
+or --debug for extra transaction output. This may require debug enpoints enable on your RPC node`,
 	Args: cobra.ExactArgs(1),
 	Run: func(cmd *cobra.Command, args []string) {
-		receipt, err := client.TransactionReceipt(context.Background(),
-			common.HexToHash(args[0]))
-		cobra.CheckErr(err)
-
-		for _, log := range receipt.Logs {
-			if log.Address == teleporterAddress {
-				logger.Info("Processing Teleporter log", zap.Any("log", log))
-
-				event, err := teleporterABI.EventByID(log.Topics[0])
-				cobra.CheckErr(err)
-
-				out, err := teleportermessenger.FilterTeleporterEvents(log.Topics, log.Data, event.Name)
-				cobra.CheckErr(err)
-				logger.Info("Parsed Teleporter event", zap.String("name", event.Name), zap.Any("event", out))
-			}
-
-			if log.Address == common.HexToAddress(warpPrecompileAddress) {
-				logger.Debug("Processing Warp log", zap.Any("log", log))
-
-				unsignedMsg, err := warp.UnpackSendWarpEventDataToMessage(log.Data)
-				cobra.CheckErr(err)
-
-				warpPayload, err := warpPayload.ParseAddressedCall(unsignedMsg.Payload)
-				cobra.CheckErr(err)
-
-				teleporterMessage, err := teleportermessenger.UnpackTeleporterMessage(warpPayload.Payload)
-				cobra.CheckErr(err)
-				logger.Info("Parsed Teleporter message",
-					zap.String("warpMessageID", unsignedMsg.ID().Hex()),
-					zap.String("teleporterMessageNonce", teleporterMessage.MessageNonce.String()),
-					zap.Any("message", teleporterMessage))
-			}
+		txHash := common.HexToHash(args[0])
+		if debug {
+			printTransaction(cmd, txHash)
+			traceTransaction(cmd, txHash)
 		}
+		checkReceipt(cmd, txHash)
 		cmd.Println("Transaction command ran successfully")
 	},
 }
 
+func checkReceipt(cmd *cobra.Command, txHash common.Hash) {
+	receipt, err := client.TransactionReceipt(context.Background(), txHash)
+	cobra.CheckErr(err)
+
+	for _, log := range receipt.Logs {
+		printTeleporterLogs(cmd, log)
+		printWarpLogs(cmd, log)
+	}
+}
+
+func printTeleporterLogs(cmd *cobra.Command, log *types.Log) {
+	if log.Address != teleporterAddress {
+		return
+	}
+
+	logJson, err := json.MarshalIndent(log, "", "  ")
+	cobra.CheckErr(err)
+
+	cmd.Println("Teleporter Log:\n" + string(logJson) + "\n")
+
+	event, err := teleporterABI.EventByID(log.Topics[0])
+	cobra.CheckErr(err)
+
+	out, err := teleportermessenger.FilterTeleporterEvents(log.Topics, log.Data, event.Name)
+	cobra.CheckErr(err)
+
+	outJson, err := json.MarshalIndent(out, "", "  ")
```

Comment:
This formats the `bytes` fields of `SourceBlockchainID` and `MessageID` as an array of ints. If we foresee human readability of these fields being valuable in the future should we do an additional post-processing to get blockchainID in cb58 form?

Reply:
I added new readable event structs that deserialize into the proper readable formats

Reply:
Looks great now!

Reply:
Rather than adding standalone types for the sole purpose of printing, we can instead extend the ABI-generated types by implementing [Stringer](https://pkg.go.dev/golang.org/x/tools/cmd/stringer) on the types of interest. We'd need to update `FilterTeleporterEvents` to return `Stringer` instead of `interface{}`, but then we can pass the returned value directly to packages that interact with `fmt` (such as `cobra`) and log using `zap` by using `zap.String`. 

Implementing `Stringer` on these types would also make them immediately usable in other logging contexts, rather than having to manually convert to the `Readable*` version, like we do in `FilterTeleporterEvents`

Reply:
I've implemented the `Stringer` interface here. It gives the same output as the previous implementation.